### PR TITLE
scale analytic jacobian for bounded variables

### LINKED ISSUE #233: analytic Jacobians not correct
see discussion at https://github.com/jjhelmus/leastsqbound-scipy/issues/6
and https://github.com/jjhelmus/leastsqbound-scipy/commit/e0ed1afbcd4c4b587e63366feffd4de8dffc1b0e

The fix should not be too hard, but slightly more complicated than for leastsqbound.

### PR DESCRIPTION
**Pull request by newville**
This PR scales the gradients calculated by user-supplied jacobian functions for bounded variables, and fixes #233.

### CODE REVIEW

File: lmfit/minimizer.py [hunk 1]
```diff
@@ -262,15 +262,26 @@ def __jacobian(self, fvars):
         analytical jacobian to be used with the Levenberg-Marquardt
 
         modified 02-01-2012 by Glenn Jones, Aberystwyth University
+        modified 06-29-2015 M Newville to apply gradient scaling
+               for bounded variables (thanks to JJ Helmus, N Mayorov
```
Comment (tacaswell) [inline lmfit/minimizer.py]: nit pick: missing `)`

File: tests/test_bounded_jacobian2.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+       c               |   %.4f    |   %.4f  |
+    ----------------------------------------------
+    ''' %  (a, b, c,
+            out1.nfev,   out2.nfev,
+            out1.chisqr, out2.chisqr,
+            out1.params['a'].value, out2.params['a'].value,
+            out1.params['b'].value, out2.params['b'].value,
+            out1.params['c'].value, out2.params['c'].value ))
+
+test_derive()
```
Comment (tacaswell) [inline tests/test_bounded_jacobian2.py]: Is this going to play nice with the automatic nose tests?

**Comment by newville** [comment]
@tacaswell  thanks -- typo fixed.  Having print statements in nose tests seem to not be a problem -- there are several throughout the lmfit tests.  They're useful for running tests by hand and suppressed when run with nosetests.

**Comment by tacaswell** [comment]
It wasn't the print I was protesting to, it was calling one of the test functions not hidden under a `__main__` test, but that file is gone now so it is a moot point.

**Pull request merged**